### ISSUE TITLE
Refactor VAP (VA Profile) components

### ISSUE THREAD

**Issue by erikphansen** [department-of-veterans-affairs/va.gov-team#9564]
The new VAP-components are based on legacy Vet360 components and much of the initial versions are simply copy-pasted from the originals. After we have a [suite of Profile 2.0 integration tests](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9563) in place We should take a stab it refining and simplifying the new components.

- [This PR](https://github.com/department-of-veterans-affairs/vets-website/pull/12860/) has a lot of comments that point out areas to address.

blocked by https://github.com/department-of-veterans-affairs/va.gov-team/issues/9563

**Comment by mshea0606** [comment]
hey @erikphansen, this issue has been unblocked.

**Comment by erikphansen** [comment]
closing this in favor of https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/16229
